ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -684,6 +694,26 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 	}
 
 	// Reconcile worker pods now
+
+	// If RayCluster is suspended, delete all worker pods and return early
+	if instance.Spec.Suspend != nil && *instance.Spec.Suspend {
+		for _, workerGroup := range instance.Spec.WorkerGroupSpecs {
+			workerPods := corev1.PodList{}
+			filterLabels = client.MatchingLabels{common.RayClusterLabelKey: instance.Name, common.RayNodeGroupLabelKey: workerGroup.GroupName}
+			if err := r.List(ctx, &workerPods, client.InNamespace(instance.Namespace), filterLabels); err != nil {
+				return err
+			}
+
+			for _, workerPod := range workerPods.Items {
```

Comment:
Dito: could it be possible to perform a single delete collection operation, e.g. with `DeleteAllOf`, instead of multiple delete ones?